src/hotspot/cpu/aarch64/sharedRuntime_aarch64.cpp

```diff
@@ -2565,20 +2566,25 @@ uint SharedRuntime::out_preserve_stack_slots() {
 // Generate a special Compile2Runtime blob that saves all registers,
 // and setup oopmap.
 //
-SafepointBlob* SharedRuntime::generate_handler_blob(address call_ptr, int poll_type) {
+SafepointBlob* SharedRuntime::generate_handler_blob(sharedStubId id, address call_ptr) {
+  assert((id >= sharedStubId::polling_page_vectors_safepoint_handler_id ||
+          id <= sharedStubId::polling_page_return_handler_id),
```

Comment:
This and all other similar assert checks depends on the order of stubs ID.
I think such checks should be in `sharedStubId` where the order is defined:
`sharedStubId::is_polling_page_id(id)`

Reply:
Yes, defining a (non-PRODUCT) validation method for this is a good idea. However, I cannot create it in sharedStubId as it is an enum class. I will put it into class SharedRuntime. I'll also create methods `is_handle_id()` and `is_throw_id()`.

I was unsure about relying on order/grouping when I wrote this. On reflection, I think it would be safer to make `is_polling_page_id()` explicitly enumerate the relevant cases. It is ok to rely on order here with SharedRuntime stubs. We are just using the declarations in stubDeclarations.hpp to define enum tags and static fields so we can define an order that groups related stubs without any further constraint. However, the current C1 stubs declaration order defines the order of generation of the stubs. Likewise, when it comes to systematizing the OptoRuntime stubs we will also want to use the declarations to drive the calls to `generate_xxx_blob()` and `generate_stub()`. So, in those cases the declaration order primarily needs to respect stub-stub dependencies.

Reply:
Latest version has non-product methods in SharedRuntime and calls them from the asserts.